lib/state_machine/state_machine.cpp

```diff
@@ -35,4 +42,39 @@ core::Result StateMachine::handleTransition(const State &state)
   return core::Result::kFailure;
 }
 
+void StateMachine::update()
+{
+  const auto nextMessage = mqtt_->getMessage();
+  if (!nextMessage) { return; }
+
+  const auto payload       = nextMessage->payload;
+  const auto message_state = string_to_state_.find((*payload)["transition"].GetString());
+
+  if (message_state != string_to_state_.end()) { handleTransition(message_state->second); }
+}
+
+void StateMachine::publishCurrentState()
+{
+  const auto state_string = stateToString(getCurrentState());
+  const auto topic        = core::MqttTopic::kTest;
+  auto message_payload    = std::make_shared<rapidjson::Document>();
+  message_payload->SetObject();
+  rapidjson::Value state_value(state_string.c_str(), message_payload->GetAllocator());
+  message_payload->AddMember("transition", state_value, message_payload->GetAllocator());
+  const core::MqttMessage::Header header{.timestamp = 0,
+                                         .priority  = core::MqttMessagePriority::kNormal};
+  const core::MqttMessage message{topic, header, message_payload};
+  mqtt_->publish(message, core::MqttMessageQos::kExactlyOnce);
+}
+
+void StateMachine::run()
+{
+  while (getCurrentState() != State::kShutdown) {
+    mqtt_->consume();
```

Comment:
this returns `kFailure` if it receives an invalid message, we should prob enter shutdown state in that case imo

Reply:
atm moment we can only transition to `kshutdown` from `kSafe`, should I add a transition from other states?